modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/AbstractAdaptiveAuthenticationTestCase.java

```diff
@@ -281,6 +281,13 @@ protected HttpResponse sendConsentGetRequest(String locationURL, CookieStore coo
 
         consentRequiredClaimsFromResponse.addAll(Utils.getConsentRequiredClaimsFromResponse(response));
         Header locationHeader = response.getFirstHeader(OAuth2Constant.HTTP_RESPONSE_HEADER_LOCATION);
+
+        if (consentRequiredClaimsFromResponse.isEmpty()){
+            String key = DataExtractUtil.extractParamFromURIFragment(locationHeader.getValue(),
+                    OAuth2Constant.SESSION_DATA_KEY_CONSENT);
+            consentRequiredClaimsFromResponse.addAll(Utils.getConsentRequiredClaimsFromDataAPI(
+                    httpClientWithoutAutoRedirections, key, userInfo, tenantInfo));
+        }
```

Comment:
Shouldn't we move the logic to Utils.getConsentRequiredClaimsFromResponse?

Reply:
resolved from https://github.com/wso2/product-is/pull/15583/commits/cfe9655a4dbf87276a4b23d637390d39d6176239